### PR TITLE
feat(discover): Use URL parameters for each visualization

### PR DESCRIPTION
**Pull request by billyvg**
Instead of using component state, use url parameters as data source for which chart gets displayed. This will be helpful when navigating from other parts of the application to a discover query.

### CODE REVIEW

File: tests/js/spec/views/organizationDiscover/result/index.spec.jsx [hunk 1]
```diff
@@ -158,9 +177,14 @@ describe('Result', function() {
 
         wrapper
           .find('ResultViewButtons')
-          .find('a')
+          .find('Link a')
           .at(1)
-          .simulate('click');
+          .simulate('click', {button: 0});
```
Comment (billyvg) [inline tests/js/spec/views/organizationDiscover/result/index.spec.jsx]: `react-router` checks for this

**Comment by lynnagara** [comment]
This seems to break toggling in discover. I'd also suggest dropping this `keepVisual` option and just doing this by default. I think it'd be good to move some of this code into a utility function that validates the visualization requested against the query rather than in `discover.jsx`

**Comment by billyvg** [comment]
@lynnagara Ah this is completely broken - I initially was comparing `location.query` instead of `location.search` in `cWRP` but it broke all the tests because they only updated `search`. I think manipulating `query` makes it a bit easier when removing/clearing query params.

The problem with always keeping visual is when you're in a by day chart and you edit your query so that it doesn't support grouping by time (e.g. can only render a Table result), how do you suggest we get around that?

**Comment by billyvg** [comment]
Updated: 6a44b31

**Comment by lynnagara** [comment]
Here's an example of how I think the persistent querystring can work - i don't think it's significantly more complicated https://github.com/getsentry/sentry/pull/11467

**Pull request closed without merging**